http-client/src/test/java/io/airlift/http/client/TestFullSmileResponseHandler.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.http.client;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.net.MediaType;
+import io.airlift.http.client.FullSmileResponseHandler.SmileResponse;
+import io.airlift.http.client.testing.TestingResponse;
+import io.airlift.json.SmileCodec;
+import io.airlift.json.SmileCodecFactory;
+import org.testng.annotations.Test;
+
+import static com.google.common.net.MediaType.PLAIN_TEXT_UTF_8;
+import static io.airlift.http.client.FullSmileResponseHandler.createFullSmileResponseHandler;
+import static io.airlift.http.client.HttpStatus.INTERNAL_SERVER_ERROR;
+import static io.airlift.http.client.HttpStatus.OK;
+import static io.airlift.http.client.testing.TestingResponse.mockResponse;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotSame;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+public class TestFullSmileResponseHandler
```

Comment:
This can also be abstracted into an `AbstractTestA__ResponseHandler`. The Codec creation can be abstracted behind factory, and specialized in the `TestResponseHandlerSmile` and `TestResponseHandlerJson`